runtime/android/core/src/org/xwalk/core/XWalkActivityDelegate.java

```diff
@@ -104,25 +151,33 @@ public void onActivateFailed() {
         mIsInitializing = false;
 
         if (mXWalkUpdater.updateXWalkRuntime()) {
-            // Set the background to screen_background_dark temporarily if the default background
-            // is null in order to avoid the visual artifacts around the alert dialog
-            Window window = mActivity.getWindow();
-            if (window != null && window.getDecorView().getBackground() == null) {
-                Log.d(TAG, "Set the background to screen_background_dark");
-                window.setBackgroundDrawableResource(android.R.drawable.screen_background_dark);
-                mBackgroundDecorated = true;
+            if (!mIsDownloadMode) {
+                // Set the background to screen_background_dark temporarily if
+                // the default background is null in order to avoid the visual
+                // artifacts around the alert dialog
+                Window window = mActivity.getWindow();
+                if (window != null &&
+                        window.getDecorView().getBackground() == null) {
+                    Log.d(TAG, "Set the background to screen_background_dark");
+                    window.setBackgroundDrawableResource(
+                            android.R.drawable.screen_background_dark);
+                    mBackgroundDecorated = true;
+                }
             }
         }
     }
 
     @Override
     public void onActivateCompleted() {
-        if (mDialogManager.isShowingDialog()) mDialogManager.dismissDialog();
-
-        if (mBackgroundDecorated) {
-            Log.d(TAG, "Recover the background");
-            mActivity.getWindow().setBackgroundDrawable(null);
-            mBackgroundDecorated = false;
+        if (!mIsDownloadMode) {
+            if (mDialogManager.isShowingDialog())
+                mDialogManager.dismissDialog();
+
+            if (mBackgroundDecorated) {
```

Comment:
Just `if (mDialogManager != null && mDialogManager.isShowingDialog())` is okay, keep other code unchanged. Because `mBackgroundDecorated` is always false, we don't need to check `mIsDownloadMode` here.